TLM/TLM/Custom/PathFinding/CustomPathManager.cs

```diff
@@ -96,8 +120,42 @@ public class CustomPathManager : PathManager {
                     Destroy(stockPathFinds[i]);
                 }
             }
+        }
+
+        public void UpdateOldPathManagerValues(PathManager stockPathManager) {
+            stockPathManager.m_simulationProfiler = m_simulationProfiler;
+            stockPathManager.m_drawCallData = m_drawCallData;
+            stockPathManager.m_properties = m_properties;
+            stockPathManager.m_pathUnitCount = m_pathUnitCount;
+            stockPathManager.m_renderPathGizmo = m_renderPathGizmo;
+            stockPathManager.m_pathUnits = m_pathUnits;
+            stockPathManager.m_bufferLock = m_bufferLock;
+
+            int n = _replacementPathFinds.Length;
+
+            Log._Debug("Creating " + n + " stock PathFind objects.");
+            PathFind[] stockPathFinds = new PathFind[n];
+
+            FieldInfo f_pathfinds = typeof(PathManager).GetField(
+                  "m_pathfinds",
+                  BindingFlags.NonPublic | BindingFlags.Instance)
+                  ?? throw new Exception("f_pathFinds is null");
 
-            InitDone = true;
+            lock(m_bufferLock) {
+                for (int i = 0; i < n; i++) {
+                    stockPathFinds[i] = gameObject.AddComponent<PathFind>();
```

Comment:
[very low priority - just worth knowing]
**Incorrect order of operations.**

`PathFind.Awake()` is caching `m_bufferLock` `m_pathUnits` references from `Singleton<PathManager>.instance` but we haven't replaced `instance` field value with stockManager yet.
Practically no issue, because we also copied those references so they should point to the same instances.